src/frontend/src/pages/scan/BarcodeInputManual.tsx

```diff
@@ -0,0 +1,64 @@
+import { t } from '@lingui/macro';
+import { ActionIcon, Group, TextInput } from '@mantine/core';
+import { getHotkeyHandler, randomId } from '@mantine/hooks';
+import { IconPlus } from '@tabler/icons-react';
+import { useState } from 'react';
+
+import { ModelType } from '../../enums/ModelType';
+
+// Scan Item
+interface ScanItem {
+  id: string;
+  ref: string;
+  data: any;
+  instance?: any;
+  timestamp: Date;
+  source: string;
+  link?: string;
+  model?: ModelType;
+  pk?: string;
+}
+
+// Region input stuff
+enum InputMethod {
+  Manual = 'manually',
+  ImageBarcode = 'imageBarcode'
+}
+
+interface inputProps {
+  action: (items: ScanItem) => void;
+}
```

Comment:
Blocking: This is defined in multiple files, there should be one shared interface

Reply:
Moved into common interface file for reuse